apps/omg_watcher/priv/repo/migrations/20190925132500_add_and_fix_timestamps.exs

```diff
@@ -0,0 +1,34 @@
+defmodule OMG.Watcher.Repo.Migrations.SetInsertedAtUpdatedAtToEpoch do
+  use Ecto.Migration
+
+  def change do
+    alter table(:ethevents) do
+      remove :inserted_at
+      remove :updated_at
+
+      timestamps([type: :timestamptz, default: "1970-01-01T00:00:00Z"])
```

Comment:
If I'm correct `default` does not fill already empty (NULL) timestamps. For me `epoch` is OK for these, but for new records I'd prefer to have `now()` instead. WDYT @jrhite :question:

Reply:
`default` does fill pre-existing rows with the default value set above. Otherwise a NOT NULL violation would occur when the new column is added.

`default` does not affect new records. new records will be set to UTC `now`.

Reply:
I disagree here, I think it's exactly the opposite:
* default does affect only new records where column's value is not specified [ecto](https://hexdocs.pm/ecto_sql/Ecto.Migration.html#add/3-options), [pg](https://www.postgresql.org/docs/9.3/ddl-default.html)
* default does NOT affect existing rows, so with combination with NOT NULL constraint would result of migration failure. [here is your fix that fixes NOT NULL problem](https://github.com/omisego/elixir-omg/pull/974/files#diff-96f62f10bc1471112c742df795e04389)

Reply:
what if we write a test and confirm what the behavior is?

Reply:
Regarding point 1: Hmm...I could share my screen and demo it for you. You may be right that `default` affects new columns where the code explicitly does not set a value. But it certainly also has the effect of backfilling pre-existing rows that existed before the new non-nullable columns. The `timestamps()` columns are indeed managed by ecto, with the caveat that calls like `insert_all()` and `update_all()` will bypass ecto's autogenerated columns.

For point 2, yes this is correct, but that's for the 2 tables I updated that already had nullable `inserted_at` and `updated_at` columns. We wanted to make them non-nullable, this the fix you reference.

Reply:
Yes my bad here.
What I quickly checked on raw postgres is when `NOT NULL` constraint is set during column addition (table exists) Default will fill pre-existing records with the value.
Inserting new records without specifying column's value will fail due to constraint.

I'm glad to learn / review what I thought I know :)